app/controllers/concerns/reauthentication_required_concern.rb

```diff
@@ -20,13 +20,17 @@ def recently_authenticated_2fa?
   private
 
   def prompt_for_second_factor
-    store_location(request.url)
+    store_location(request.path)
     user_session[:context] = 'reauthentication'
 
     redirect_to login_two_factor_options_path
   end
 
   def store_location(url)
-    user_session[:stored_location] = url
+    if request.method != :get
```

Comment:
this is causing issues with
./spec/features/two_factor_authentication/change_factor_spec.rb